src/widget/wtracktableview.cpp

```diff
@@ -255,12 +264,18 @@ void WTrackTableView::loadTrackModel(QAbstractItemModel *model) {
      * there's no need to exchange the headers
      * this will cause a small GUI freeze
      */
+    WTrackTableViewHeader* oldHeader =
+            dynamic_cast<WTrackTableViewHeader*>(horizontalHeader());
     if (getTrackModel() == trackModel) {
-        // Re-sort the table even if the track model is the same. This triggers
-        // a select() if the table is dirty.
-        doSortByColumn(horizontalHeader()->sortIndicatorSection());
+        // Call select() on the table so it refreshes if it's dirty.
+        getTrackModel()->select();
```

Comment:
I'm also unsure about this change... we definitely can't just force-sort on the existing column because that could be the wrong column

Reply:
Doesn't the old header state hold the old sorting column? What can go wrong here?

Reply:
The problem only occurs when we switch between crates -- the crate feature changes crate ID before it gets the signal to save the header.